docs/how-to/h-upgrade.md

```diff
@@ -0,0 +1,115 @@
+# How to upgrade between minor versions
+
+> **:warning: WARNING:** this is currently a feature in "edge" or "beta" risks channels. We do NOT recommend to use features in production that are not in the `stable` channel! Also, contact [Canonical Data Platform team](https://chat.charmhub.io/charmhub/channels/data-platform) if you are interested in the topic.
+
+Charm upgrades allow admin to upgrade both operator code (e.g. the revision used by the charm) and/or the workload version. Note that since the charm code pin a particular version of the workload, a charm upgrade may or may not involve also a workload version upgrade. In general, the following guide only applies for in-place upgrade that involve at most minor version upgrade of Kafka workload, e.g. between Kafka 3.4.x to 3.5.x. Major workload upgrades are generally **NOT SUPPORTED** and they should be carried out using out-of-place migrations. Please refer to the how-to guide about cluster migration [here](/t/charmed-kafka-how-to-cluster-migration/10951) for more information on how this can be achieved.
```

Comment:
*allow admins

Reply:
*charm code pins

Reply:
**nitpick:** involve - at most - ...

Reply:
**suggestion:** 'out-of-place' sounds... Out of place (ha!). 

Maybe say something evocative of 'full cluster to cluster migration'